tealium/collectors/attribution/TealiumAdClient.swift

```diff
@@ -25,7 +36,54 @@ public class TealiumAdClient: TealiumAdClientProtocol {
     }
 
     public func requestAttributionDetails(_ completionHandler: @escaping ([String: NSObject]?, Error?) -> Void) {
-        adClient.requestAttributionDetails(completionHandler)
+        if #available(iOS 14.3, *) {
+            if let adAttributionToken = try? AAAttribution.attributionToken() {
+                guard let url = URL(string: "https://api-adservices.apple.com/api/v1/") else {
+                    completionHandler(nil, AdServiceErrors.invalidUrl)
+                    return
+                }
+                let request = NSMutableURLRequest(url: url)
+                request.httpMethod = "POST"
+                request.setValue("text/plain", forHTTPHeaderField: "Content-Type")
+                request.httpBody = adAttributionToken.data(using: .utf8)
+                let task = URLSession.shared.dataTask(with: request as URLRequest, completionHandler: { data, _, error in
+                    if let error = error {
+                        completionHandler(nil, error)
+                        return
+                    }
+                    do {
+                        guard let data = data else {
+                            completionHandler(nil, AdServiceErrors.nilData)
+                            return
+                        }
+                        if let jsonResponse = try JSONSerialization.jsonObject(with: data, options: .fragmentsAllowed) as? [String: NSObject] {
```

Comment:
Why the fragmentsAllowed?
Also, did we evaluate having a DTO with Codable support to map to the persistentAppData instead of a dictionary? It probably makes (at least the new API) more readable

Reply:
I believe fragmentsAllowed is needed right? Doesn't fragmentsAllowed allow non dictionary / array top level keys? In this version of the api the top level contains all they key value pairs.

Reply:
From what i see here: https://developer.apple.com/documentation/ad_services/aaattribution/3697093-attributiontoken#3697458
They return a normal JSON:
```
{
  "attribution": true,
  "orgId": 40669820,
  "campaignId": 542370539,
  "conversionType": "downloads",
  "adGroupId": 542317095,
  "countryOrRegion": "US",
  "keywordId": 87675432,
  "adId": 542317136
}
```
Is this not the case?

I think that that key is needed if the payload returns something like:
```
3
```
Then the fragmentsAllowed would correctly json decode the 3 into the actual number 3 as it is a fragment of a bigger JSON.